src/patternfly/components/ModalBox/modal-box.scss

```
@@ -61,7 +61,7 @@
   --#{$modal-box}__header--body--PaddingBlockStart: var(--pf-t--global--spacer--sm);
 
   // Close
-  --#{$modal-box}__close--InsetBlockStart: var(--#{$modal-box}__header--PaddingBlockStart);
+  --#{$modal-box}__close--InsetBlockStart: var(--pf-t--global--spacer--md);
```

WDYT about just offsetting the close button by its own padding? It's the same spacer as `var(--pf-t--global--spacer--md)` (16px/1rem) but will still be aligned if either the modal header's padding or button padding changes.

```suggestion
  --#{$modal-box}__close--InsetBlockStart: calc(var(--#{$modal-box}__header--PaddingBlockStart) - var(--pf-t--global--spacer--control--vertical--default));
```